### PR TITLE
fix(rspack):optimize the builtin:swc-loader matching calculation

### PR DESCRIPTION
**Pull request by easy1090**
## Summary
fix(rspack):optimize the builtin:swc-loader matching calculation
## Related Links
closes:#166 

## Checklist

- [x] I have added changeset via `pnpm run change`.
- [ ] I have updated the documentation.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: packages/core/src/build-utils/build/utils/loader.ts [hunk 1]
```diff
@@ -167,6 +167,82 @@ export function mapEachRules<T extends Plugin.BuildRuleSetRule>(
...
+      };
+      return appendRules(_rule, 0);
+    }
+
+    if (rule.use) {
+      if (Array.isArray(rule.use)) {
+        const _index = findIndex(
+          rule.use,
+          (_r) =>
+            (typeof _r === 'object' &&
```
Comment (chenjiahan) [inline packages/core/src/build-utils/build/utils/loader.ts:203]: It is recommended to extract some small helper functions to increase code readability

File: packages/core/src/build-utils/build/utils/loader.ts [hunk 2]
```diff
@@ -167,6 +167,82 @@ export function mapEachRules<T extends Plugin.BuildRuleSetRule>(
...
+              typeof _r?.loader === 'string' &&
+              _r.loader === loaderName) ||
+            (typeof _r === 'string' && _r === loaderName),
+        );
+        if (_index > -1) {
+          return appendRules(rule, _index);
+        }
+      } else if (typeof rule.use === 'object' && !Array.isArray(rule.use)) {
```
Comment (SoonIter) [inline packages/core/src/build-utils/build/utils/loader.ts:211]: maybe rule.use is a function type?

Comment (easy1090) [reply packages/core/src/build-utils/build/utils/loader.ts]: > maybe rule.use is a function type?

fixed

File: packages/core/src/build-utils/build/utils/loader.ts (hunk 1 above)
Comment (easy1090) [reply packages/core/src/build-utils/build/utils/loader.ts]: > It is recommended to extract some small helper functions to increase code readability

fixed

**Review by chenjiahan — APPROVED** [review]

**Pull request merged**